### PR TITLE
Add discord account linking

### PR DESCRIPTION
**Pull request by tsunyoku**
Depends on https://github.com/osuAkatsuki/mysql-database/pull/17.

### CODE REVIEW

File: app/v1/connections.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	var tokenResp struct {
+		AccessToken string `json:"access_token"`
+	}
+
+	if err := json.Unmarshal(resp.Body(), &tokenResp); err != nil {
+		md.Err(err)
+		return Err500
+	}
+
+	userResp, err := client.R().
```
Comment (cmyui) [inline app/v1/connections.go:53]: is this an ephemeral/short-lived client? Is that `setAuthToken()` going to have state last more than just one request?

File: app/start.go [hunk 2]
```diff
@@ -104,6 +104,8 @@ func Start(dbO *sqlx.DB) *fhr.Router {
 		r.Method("/api/v1/countries", v1.CountriesGET)
 		r.Method("/api/v1/hypothetical-rank", v1.HypotheticalRankGET)
 
+		r.Method("/api/v1/discord/callback", v1.DiscordCallbackGET, common.PrivilegeReadConfidential)
```
Comment (cmyui) [inline app/start.go:107]: PrivilegeReadConfidential = auth'd normal user privs?

**Review by cmyui — APPROVED** [review]

File: app/v1/connections.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+	if err := json.Unmarshal(userResp.Body(), &discordUser); err != nil {
+		md.Err(err)
+		return Err500
+	}
+
+	md.DB.Exec("UPDATE users SET discord_account_id = ? WHERE id = ?", discordUser.ID, md.ID())
+
+	md.Ctx.Redirect("https://akatsuki.gg", 301)
+	return common.SimpleResponse(301, "")
```
[the remark this answers is not in the document]
Comment (tsunyoku) [reply app/v1/connections.go:74]: I'm unsure if/how we can do this from the API

File: app/start.go (hunk 2 above)
Comment (tsunyoku) [reply app/start.go:107]: yep

File: app/v1/connections.go (hunk 1 above)
Comment (tsunyoku) [reply app/v1/connections.go:53]: `SetAuthToken` is only needed for this one request, but if we re-use the client then it'll have an older token pre-populated which seems less than ideal

Comment (cmyui) [reply app/v1/connections.go:53]: Ye make sure it gets reset after each request -- should be ephemeral/single-request state. That'd be a really bad security hole lol

File: app/v1/connections.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (cmyui) [reply app/v1/connections.go:74]: simplest form, redirect w/ `?discordSuccess=true`

**Pull request merged**